### PR TITLE
Emit errors for collection assignments

### PR DESCRIPTION
**Pull request by nirinchev**
## Description

Fixes https://github.com/realm/realm-dotnet/issues/3455

##  TODO

* [x] Changelog entry
* [x] Tests (if applicable)

### CODE REVIEW

File: Tests/SourceGenerators/Realm.SourceGenerator.Tests/ComparisonTests.cs [hunk 1]
```diff
@@ -41,19 +41,28 @@ public async Task ComparisonTest(string filename, params string[] classNames)
...
-        [TestCase("RealmObjectAndEmbeddedObjectClass")]
-        [TestCase("UnsupportedIndexableTypes")]
-        [TestCase("UnsupportedPrimaryKeyTypes")]
-        [TestCase("UnsupportedRequiredTypes")]
-        [TestCase("NestedClassWithoutPartialParent")]
-        [TestCase("NullableErrorClass")]
-        [TestCase("IgnoreObjectNullabilityClass")]
-        [TestCase("UnsupportedBacklink", "UnsupportedBacklink", "BacklinkObj")]
-        public async Task ErrorComparisonTest(string filename, params string[] classNames)
+        public static object[][] ErrorTestCases = Directory.GetFiles(_errorClassesPath)
```
Comment (papafe) [inline Tests/SourceGenerators/Realm.SourceGenerator.Tests/ComparisonTests.cs:44]: I understand the sentiment, but you have to admit it's kinda ugly 😁

Comment (nirinchev) [reply Tests/SourceGenerators/Realm.SourceGenerator.Tests/ComparisonTests.cs:44]: Agree, but we clearly forget to add test cases there - the CollectionError tests were missing and I was wondering for quite some time why nothing is failing when I changed my code 😅

Comment (nirinchev) [reply Tests/SourceGenerators/Realm.SourceGenerator.Tests/ComparisonTests.cs]: I reworked the test discovery - let me know if now it's prettier 😜

Comment (papafe) [reply Tests/SourceGenerators/Realm.SourceGenerator.Tests/ComparisonTests.cs]: Regexes ftw it seems! It looks better I think ;)

**Pull request merged**